dataclients/kubernetes/kube.go

```diff
@@ -957,6 +972,29 @@ func (c *Client) ingressToRoutes(state *clusterState) ([]*eskip.Route, error) {
 	return routes, nil
 }
 
+func defaultFiltersOf(i *ingressItem, defaultFilters map[resourceId]string) (string, bool) {
+	var filters []string
+	services := make(map[string]struct{})
+	for _, r := range i.Spec.Rules {
+		for _, p := range r.Http.Paths {
+			services[p.Backend.ServiceName] = struct{}{}
+		}
+	}
+
+	for s := range services {
+		if val, ok := defaultFilters[resourceId{name: s, namespace: i.Metadata.Namespace}]; ok {
+			filters = append(filters, val)
```

Comment:
This loop is against my intuition. I currently can't prove, but I think this function is doing the wrong thing and is called from the wrong place in the execution path.

Reply:
refactored to add the default filters per route (instead of doing it per ingress)